Mavlink-router failing to stop

```bash
def run_common_routing_tests(
        router: AbstractRouter, allowed_output_types: List[EndpointType], allowed_master_types: List[EndpointType]
    ) -> None:
        allowed_output_endpoints = [e for e in OUTPUT_ENDPOINTS if e.connection_type in allowed_output_types]
        for endpoint in allowed_output_endpoints:
            assert router.add_endpoint(endpoint), f"Failed to add endpoint {endpoint}."
        assert router.endpoints() == allowed_output_endpoints, "Endpoint list does not match."
    
        for endpoint in [e for e in OUTPUT_ENDPOINTS if not e.connection_type in allowed_output_types]:
            with pytest.raises(NotImplementedError):
                router.add_endpoint(endpoint)
    
        for endpoint in [e for e in MASTER_ENDPOINTS if e.connection_type in allowed_master_types]:
            assert router.start(endpoint), f"Failed to start MAVLinkRouter with endpoint {endpoint}."
            assert router.is_running(), "MAVLinkRouter is not running after start."
            assert router.exit(), "MAVLinkRouter could not stop."
>           assert not router.is_running(), "MAVLinkRouter is not stopping after exit."
E           AssertionError: MAVLinkRouter is not stopping after exit.
E           assert not True
E            +  where True = <bound method AbstractRouter.is_running of <mavlink_proxy.MAVLinkRouter.MAVLinkRouter object at 0x7f6b38146d00>>()
E            +    where <bound method AbstractRouter.is_running of <mavlink_proxy.MAVLinkRouter.MAVLinkRouter object at 0x7f6b38146d00>> = <mavlink_proxy.MAVLinkRouter.MAVLinkRouter object at 0x7f6b38146d00>.is_running

core/services/ardupilot_manager/mavlink_proxy/test_all.py:126: AssertionError
```

Calling `router.is_running()` after `router.exit()` is apparently returning True, as can be seen by this resport from pytest.

--- a/core/services/ardupilot_manager/mavlink_proxy/test_all.py
+++ b/core/services/ardupilot_manager/mavlink_proxy/test_all.py
@@ -116,6 +116,7 @@ def test_mavlink_router(valid_output_endpoints: List[Endpoint], valid_master_end
     )
 
 
+@pytest.mark.timeout(10)
 def run_common_routing_tests(
     router: AbstractRouter,
     allowed_output_types: List[EndpointType],
@@ -144,6 +145,9 @@ def run_common_routing_tests(
         assert router.start(endpoint), f"Failed to start {router.name()} with endpoint {endpoint}."
         assert router.is_running(), f"{router.name()} is not running after start."
         assert router.exit(), f"{router.name()} could not stop."
+        while router.is_running():
+            pass
+        assert not router.is_running(), f"{router.name()} is not stopping after exit."
 
     unallowed_master_endpoints = list(
         filter(lambda endpoint: endpoint not in allowed_master_endpoints, master_endpoints)

